src/Stratis.FederatedPeg.Tests/TransactionBuilderTests.cs

```diff
@@ -128,43 +121,171 @@ public void BuildTransactionBuildsDeterministicTransactions()
+
+            var recipient2 = new List<Recipient.Recipient>()
+            {
+                new Recipient.Recipient
+                {
+                    Amount = Money.COIN * 70,
+                    ScriptPubKey = PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(new Key().PubKey.Hash)
+                }
+            };
+
+            string opReturnData2 = blockHeight.ToString();
+
+            // Build the multisig transaction template.
+            var multiSigContext2 = new TransactionBuildContext(recipient2, opReturnData: Encoding.UTF8.GetBytes(opReturnData2))
+            {
+                TransactionFee = Money.Coins(0.01m),
+                MinConfirmations = 0,
+                Shuffle = false,
+                MultiSig = multiSigAddress,
+                IgnoreVerify = true,
+                Sign = false
+            };
+
+            // Build the transaction.
+            Transaction transaction2 = this.federationTransactionHandler.BuildTransaction(multiSigContext2);
+
+            // Transactions inputs.
+            Assert.Equal(2, transaction2.Inputs.Count);
+            Assert.Equal(transaction1.GetHash(), transaction2.Inputs[0].PrevOut.Hash);
+            Assert.Equal((uint)0, transaction2.Inputs[0].PrevOut.N);
+            Assert.Equal((uint256)2, transaction2.Inputs[1].PrevOut.Hash);
+            Assert.Equal((uint)0, transaction2.Inputs[1].PrevOut.N);
+
+            // Transaction outputs.
+            Assert.Equal(3, transaction2.Outputs.Count);
+
+            // Transaction output value - change.
+            Assert.Equal(new Money(9.98m, MoneyUnit.BTC), transaction2.Outputs[0].Value);
+            Assert.Equal(multiSigAddress.ScriptPubKey, transaction2.Outputs[0].ScriptPubKey);
+
+            // Transaction output value - recipient 2.
+            Assert.Equal(new Money(70m, MoneyUnit.BTC), transaction2.Outputs[1].Value);
+            Assert.Equal(recipient2[0].ScriptPubKey, transaction2.Outputs[1].ScriptPubKey);
+
+            // Transaction output value - op_return.
+            Assert.Equal(new Money(0m, MoneyUnit.BTC), transaction2.Outputs[2].Value);
+            Assert.Equal(opReturnData2, new OpReturnDataReader(this.loggerFactory, this.network).GetString(transaction2, out OpReturnDataType dummy2));
+
+            // Reserve UTXO's spent by transaction 2.
+            this.federationWalletManager.ProcessTransaction(transaction2);
+
+            // All the input UTXO's should be present in spending details of the multi-sig address.
+            Assert.True(this.SanityCheck(transaction2));
+
+        }
+
+        /// <summary>
+        /// Verifies that the transaction's inout UTXO's have been reserved by the wallet.
+        /// </summary>
+        /// <param name="transaction">The transaction to check.</param>
+        /// <returns><c>True</c> if all's well and <c>false</c> otherwise.</returns>
+        private bool SanityCheck(Transaction transaction)
```

Comment:
Is it possible to reuse the 
public async Task SanityCheckAsync() or its private Sync version ?

Reply:
Done